extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/FormDataOutputMapperGenerator.java

```diff
@@ -0,0 +1,251 @@
+     */
+    public static boolean isReturnTypeCompatible(ClassInfo returnTypeClassInfo, IndexView index) {
+        // go up the class hierarchy until we reach Object
+        ClassInfo currentClassInHierarchy = returnTypeClassInfo;
+        while (true) {
+            List<FieldInfo> fields = currentClassInHierarchy.fields();
+            for (FieldInfo field : fields) {
+                if (Modifier.isStatic(field.flags())) { // nothing we need to do about static fields
+                    continue;
+                }
+
+                if (field.annotation(REST_FORM_PARAM) != null || field.annotation(FORM_PARAM) != null) {
+                    // Found either @RestForm or @FormParam in returning class, it's compatible.
+                    return true;
+                }
+            }
+
+            DotName superClassDotName = currentClassInHierarchy.superName();
+            if (superClassDotName.equals(DotNames.OBJECT_NAME)) {
+                break;
+            }
+            ClassInfo newCurrentClassInHierarchy = index.getClassByName(superClassDotName);
+            if (newCurrentClassInHierarchy == null) {
+                printWarningMessageForMissingJandexIndex(currentClassInHierarchy, superClassDotName);
+                break;
+            }
+
+            currentClassInHierarchy = newCurrentClassInHierarchy;
+        }
+
+        // if we reach this point then the returning type is not compatible.
+        return false;
+    }
+
+    /**
+     * Generates a class that map a Pojo into {@link PartItem} that is then used by {@link MultipartMessageBodyWriter}.
+     *
+     * <p>
+     * For example for a pojo like:
+     *
+     * <pre>
+     * public class FormData {
+     *
+     *     &#64;RestForm
+     *     &#64;PartType(MediaType.TEXT_PLAIN)
+     *     private String text;
+     *
+     *     &#64;RestForm
+     *     &#64;PartType(MediaType.APPLICATION_OCTET_STREAM)
+     *     public File file;
+     *
+     *     public String getText() {
+     *         return text;
+     *     }
+     *
+     *     public void setText(String text) {
+     *         this.text = text;
+     *     }
+     *
+     *     public File getFile() {
+     *         return file;
+     *     }
+     *
+     *     public void setFile(File file) {
+     *         this.file = file;
+     *     }
+     * }
+     * </pre>
+     *
+     * <p>
+     *
+     * The generated mapper would look like:
+     * public class FormData_generated_mapper implements MultipartOutputInjectionTarget {
+     * public FormDataOutput mapFrom(Object var1) {
+     * FormDataOutput var2 = new FormDataOutput();
+     * FormData var4 = (FormData)var1;
+     * File var3 = var4.data;
+     * MultipartSupport.addPartItemToFormDataOutput(var2, "file", "application/octet-stream", var3);
+     * File var5 = var4.text;
+     * MultipartSupport.addPartItemToFormDataOutput(var2, "text", "text/plain", var5);
+     * return var2;
```

Comment:
This could use better formatting as it's not readable

Reply:
Ups, this is wrong. Thanks for spotting. Fixed.